File: src/main/java/io/github/tanguygab/playerlistexpansion/CachedList.java
package io.github.tanguygab.playerlistexpansion;

import org.bukkit.OfflinePlayer;

import java.util.List;
import java.util.function.Function;

public class CachedList {

    private long lastUpdate;
    private final Function<OfflinePlayer, List<String>> fun;
    private List<String> list;

    public CachedList(Function<OfflinePlayer, List<String>> fun) {
        this.fun = fun;
    }

    public List<String> getList(OfflinePlayer player) {
        long now = System.currentTimeMillis();
        System.out.println(now - lastUpdate > PlayerListExpansion.get().updateCooldown);
        if (now - lastUpdate > PlayerListExpansion.get().updateCooldown) {
            list = fun.apply(player);
            lastUpdate = now;
        }
        return list;
    }
}

File: src/main/java/io/github/tanguygab/playerlistexpansion/CachedList.kt
package io.github.tanguygab.playerlistexpansion

import org.bukkit.OfflinePlayer
import java.util.function.BiFunction

class CachedList(private val function: BiFunction<OfflinePlayer?, String, List<String>>) {
    private var lastUpdate: Long = 0
    private var list = listOf<String>()

    fun getList(player: OfflinePlayer?, format: String): List<String> {
        val now = System.currentTimeMillis()
        if (now - lastUpdate > PlayerListExpansion.get().updateCooldown) {
            list = function.apply(player, format)
            lastUpdate = now
        }
        return list
    }
}
